tests/rptest/tests/acls_test.py

```diff
@@ -462,10 +462,10 @@ def test_upgrade_sasl(self):
             pass_w_super_user=True,
             err_msg='check_permissions failed before upgrade')
 
-        self.installer.install(self.redpanda.nodes, RedpandaInstaller.HEAD)
```

Comment:
I think job 1 is to stabilize the definition of the test on `dev`, get it all squared away + then think about backporting the test and removing it from dev.  (in my mind that the first part on dev is before the holidays, the second part is after)

Reply:
Related to this, we need to figure out how to write these test while introducing a feature. One way would be to install HEAD, and after the release apply a change to use the correct tuple. Another one would be to wire latest_for_line(<current feature release>) to return HEAD

Reply:
Good point, I think we can cover that with latest_for_line anyway, because we tag the tip of dev with the next expected release + `-dev` (e.g. the current tip of dev is something like `v23.1.0-dev-617-ge2d565298`)

Reply:
@jcsp can you explain? I see a tag `v23.1.0-dev` but it points to this commit
`2022-11-11 16:41 +0000 John Spray           M─┐ [HEAD] <v23.1.0-dev> Merge pull request #7251 from jcsp/github-template-23.1.x
`

Reply:
I was thinking that we would have test code that would look for a released 23.1.x packaged if one exists, and if none exists and HEAD version matches v23.1.0-dev, then we would use HEAD for the test.  That way we get a test that works on `dev`, and then cuts over to using released packages as soon as they exist.

Reply:
ENABLE_GIT_HASH=OFF is the one you want for upgrade tests to work locally

Reply:
I think that would work on `dev` but it isn't so straightforward for older branches. If we implement search for builds in the current line and use HEAD if it doesn't exist, we'll need to opt out of that when we're not on a x.x.0 version (e.g. consider what happens if we're on v22.3.x, where the git version is v22.3.9; if we're not careful we may end up just upgrading to the released v22.3.9 rather than the tip of v22.3.x)

Reply:
@andrwng the new code should select HEAD if requesting the current release line.
The RedpandaInstaller.install method is modified so that - pretending we already had a release v23.1.0 - 
install(..., (23, 1, 0)) will install the release from github, and install(..., (23, 1)) will install dev tip. 

once the bugs are fixed, obviously.

Reply:
It's good that I'm writing this code at the end of the year, I caught a bug In this pr that I would have probably missed until the next roll of the year